### PR TITLE
chore: bump dependencies

### PR DESCRIPTION
**Pull request by zepatrik**
```
github.com/auth0/go-jwt-middleware v1.0.1 => v2.2.1
github.com/docker/docker v20.10.27+incompatible => v27.0.3+incompatible
github.com/form3tech-oss/jwt-go v3.2.5+incompatible => github.com/golang-jwt/jwt/v5 v5.2.1
github.com/ory/dockertest/v3 v3.10.1-0.20240619125955-3328cf9343b8 => v3.10.1-0.20240704115616-d229e74b748d
github.com/rs/cors v1.8.2 => v1.11.0
```

### CODE REVIEW

File: corsx/middleware_test.go [hunk 1]
```diff
@@ -32,7 +32,7 @@ func TestContextualizedMiddleware(t *testing.T) {
 		require.NoError(t, err)
 		req.Header.Set("Origin", origin)
 		req.Header.Set("Access-Control-Request-Method", "DELETE")
-		req.Header.Set("Access-Control-Request-Headers", "origin")
+		req.Header.Set("Access-Control-Request-Headers", "")
```
Comment (zepatrik) [inline corsx/middleware_test.go:35]: Apparently it is not a thing that the browser sets the origin header here:
https://github.com/rs/cors/issues/151

File: jwtmiddleware/middleware_test.go [hunk 2]
```diff
@@ -86,48 +87,49 @@ func TestSessionFromRequest(t *testing.T) {
...
+			token:               "",
+			expectedStatusCode:  401,
+			expectedErrorReason: "Authorization header format must be Bearer {token}",
 		},
 		// token without kid
 		{
-			token:              "eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwiZXhwIjo5OTk5OTk5OTk5LCJzZXNzaW9uIjp7ImlkZW50aXR5Ijp7ImlkIjoiMTIzNDU2Nzg5MCJ9fX0.j0SgjC21nhkNP2QX0uE-I4wDYYRYlZq9wqGeDhrbplkKGW4BOjW5Sk0XFFbqrx68hQYz23QvYOYW5avUBzTjPxHwVqB1HPv6M5P2wHvRn7ZvAyhz83fmJMnBRNBOz1MfjxnEgkwfcVbNqsW2y37kRdZfveBlAzSfuPJV8Rkb4wlBbEGUwoCk78j8zcD_dcYFfXbt7uXz_tscScoIOg959Rmwr2E1XqRNy2qWLKSImwo8athdEEE-byLYytg6mgM02bmEQk2dyd5W2MmqG_4UaiBru6Bf9-drqExHDGUyndnAKi_uvF_131_LkPxy6H5Hu_YfZgSE5hXUbRsBzU-gbY5aV5FSn855PnRDyS_lFnBEn-0vcCIMmxbdfhqyKtFPmFHdSO1YsGruhqYaOLOlTVzThP-1XJSpgMKXHXW35c52zB9AaTV-0ETICvZ_OjZM_uzdWeb6PQmFsztcwdO-9C70yR3_HdcjljvnQ4XHs9ho_3_V57fcbW3uQCTq0TRbwD0AXpkVOvKJqaP1yEXYLKSNpGL2MMkuY-i3k6wTZMTV1280TqbJcSpY5n6WoWJnjoZ08BwBQDfX8AUsKk-D71wJbONqmLo5YnmrS-1gHR3bKCfuUzDdvensLXYJwSHg3ae_qE5VxscRhT_p2odeE8JgQBhd0d6765YBAP93F1c",
-			expectedStatusCode: 401,
-			expectedResponse:   "jwt from authorization HTTP header is missing value for \"kid\" in token header",
```
Comment (zepatrik) [inline jwtmiddleware/middleware_test.go:110]: The previous errors exposed here were not really something a legitimate user would run into, and they could not fix it either. The generic version is fine IMO.

File: jwtmiddleware/middleware_test.go [hunk 3]
```diff
@@ -140,11 +142,15 @@ func TestSessionFromRequest(t *testing.T) {
 			require.NoError(t, err)
 			defer res.Body.Close()
 
-			body, err := ioutil.ReadAll(res.Body)
+			body, err := io.ReadAll(res.Body)
 			require.NoError(t, err)
 
 			assert.Equal(t, tc.expectedStatusCode, res.StatusCode, string(body))
-			assert.Equal(t, tc.expectedResponse, strings.TrimSpace(string(body)))
+			assert.Equal(t, tc.expectedErrorReason, gjson.GetBytes(body, "error.reason").String())
```
Comment (zepatrik) [inline jwtmiddleware/middleware_test.go:149]: We now always return JSON errors.

File: go.mod [hunk 4]
```diff
@@ -6,17 +6,16 @@ toolchain go1.22.2
...
 	github.com/bmatcuk/doublestar/v2 v2.0.4
 	github.com/bradleyjkemp/cupaloy/v2 v2.8.0
 	github.com/cenkalti/backoff/v4 v4.3.0
 	github.com/cockroachdb/cockroach-go/v2 v2.3.5
 	github.com/dgraph-io/ristretto v0.1.1
-	github.com/docker/docker v20.10.27+incompatible
+	github.com/docker/docker v27.0.3+incompatible
 	github.com/evanphx/json-patch/v5 v5.6.0
 	github.com/fatih/structs v1.1.0
-	github.com/form3tech-oss/jwt-go v3.2.5+incompatible
```
Comment (zepatrik) [inline go.mod:19]: Moved to a maintained library.

File: go.mod [hunk 5]
```diff
@@ -6,17 +6,16 @@ toolchain go1.22.2
...
-	github.com/auth0/go-jwt-middleware v1.0.1
+	github.com/auth0/go-jwt-middleware/v2 v2.2.1
 	github.com/avast/retry-go/v4 v4.3.0
 	github.com/bmatcuk/doublestar/v2 v2.0.4
 	github.com/bradleyjkemp/cupaloy/v2 v2.8.0
 	github.com/cenkalti/backoff/v4 v4.3.0
 	github.com/cockroachdb/cockroach-go/v2 v2.3.5
 	github.com/dgraph-io/ristretto v0.1.1
-	github.com/docker/docker v20.10.27+incompatible
+	github.com/docker/docker v27.0.3+incompatible
```
Comment (aeneasr) [inline go.mod:16]: This could break dockertest

File: jwtmiddleware/middleware.go [hunk 6]
```diff
@@ -89,35 +101,57 @@ func NewMiddleware(
...
-					return nil, errors.New(`jwt from authorization HTTP header is missing value for "kid" in token header`)
-				} else if kid, ok := raw.(string); !ok {
-					return nil, fmt.Errorf(`jwt from authorization HTTP header is expecting string value for "kid" in tokenWithoutKid header but got: %T`, raw)
-				} else if k, err := jc.GetKey(kid); err != nil {
-					return nil, err
-				} else {
-					return k.Key, nil
-				}
+		jm: jwtmiddleware.New(
+			func(ctx context.Context, rawToken string) (any, error) {
```
Comment (aeneasr) [inline jwtmiddleware/middleware.go:105]: Is debug missing now?

File: jwtmiddleware/middleware_test.go (hunk 2 above)
Comment (aeneasr) [reply jwtmiddleware/middleware_test.go:110]: I kind of disagree, it makes debugging so much more difficult!

File: jwtmiddleware/middleware.go (hunk 6 above)
Comment (zepatrik) [reply jwtmiddleware/middleware.go:105]: Yes, the debug option does not exist anymore. Instead, the errors returned should have more context.

File: go.mod (hunk 5 above)
Comment (zepatrik) [reply go.mod:16]: I can also use the same version as dockertest.

**Pull request merged**